python/lsst/analysis/tools/atools/metadataMetrics.py

```diff
@@ -41,9 +41,11 @@ class MetadataMetricTool(AnalysisTool):
         default=None,
     )
 
-    subTaskName = Field[str](
-        doc="The name of the subtask to extract metadata from. "
-        "If None, the entire task metadata will be used.",
+    subTaskNames = DictField[str, str](
+        doc="The names of the subtasks to extract metadata from. "
+        "If the metric name is identified as one of the keys, then "
+        "the corresponding value is taken as the subTask metadata "
+        "from which to extract metadata.",
```

Comment:
The old comment is I think still also helpful: `"If None, the entire task metadata will be used."`

Reply:
I think now that the old comment was incorrect. I now think that whether it uses the whole metedata depends on whether the metadata is saved as JSON or YAML. Not ideal, since one often doesn't know how metadata is saved since we let the butler take care of that. I've not looked into it deeply, though, so that's just my hunch atm.

Either way, I know that in some instances it doesn't search the whole metadata if subTaskNames is None, so this system is safer.